python/requirements/ml/requirements_dl.txt

```diff
@@ -4,12 +4,13 @@
 tensorflow==2.6.2
 tensorflow-probability==0.14.1
 
-# If you make changes to the torch versions below, please also make the corresponding changes to `requirements_ml_docker.txt`!
+# If you make changes to the versions below, please also make the corresponding changes to `requirements_ml_docker.txt`!
 
-torch==1.9.0;sys_platform=="darwin"
-torchvision==0.10.0;sys_platform=="darwin"
-
-# On non-OSX machines only install CPU version of torch and torchvision
--f https://download.pytorch.org/whl/torch_stable.html
-torch==1.9.0+cpu;sys_platform!="darwin"
-torchvision==0.10.0+cpu;sys_platform!="darwin"
+--extra-index-url https://download.pytorch.org/whl/cpu  # for CPU versions of torch, torchvision
+--find-links https://data.pyg.org/whl/torch-1.9.0+cpu.html  # for CPU versions of torch-scatter, torch-sparse
+torch==1.9.0
+torchvision==0.10.0
+torch-scatter==2.0.9
+torch-sparse==0.6.12
```

Comment:
it [looks](https://buildkite.com/ray-project/ray-builders-pr/builds/39918#018236f2-cba3-43ef-af51-39a5777fef19/160-1873) like pip won't overwrite these torch-scatter and torch-sparse CPU versions with the corresponding GPU versions, because the wheel filenames are identical (missing the plus notation), just living under a different url.

since CI was green with only the GPU link, I will attempt to specify the GPU link here, and remove the (then) duplicate entries from `requirements_ml_docker.txt`

Reply:
3a71677

Reply:
d'oh, wheels for torch_sparse/torch_scatter + cuda + mac are [not available](https://buildkite.com/ray-project/ray-builders-pr/builds/40006#018239c3-9750-460e-b8e5-32446c11ceeb/1408-1590).

looks like 3a71677 needs to be reverted, as the GPU install command won't be run on Mac

Reply:
reverted: [fabcfbf](https://github.com/ray-project/ray/pull/26072/commits/fabcfbf495be78356145a0271a7a2aeb53277640)